frontend/src/Frontend/Transfer.hs

```diff
@@ -0,0 +1,235 @@
+{-# LANGUAGE FlexibleContexts           #-}
+{-# LANGUAGE LambdaCase                 #-}
+{-# LANGUAGE OverloadedStrings          #-}
+{-# LANGUAGE RecursiveDo                #-}
+{-# LANGUAGE ScopedTypeVariables        #-}
+{-# LANGUAGE TupleSections              #-}
+{-# LANGUAGE TypeApplications           #-}
+
+module Frontend.Transfer where
+
+------------------------------------------------------------------------------
+import qualified Data.Aeson as A
+import           Data.Foldable
+import           Data.Functor
+import           Data.Scientific
+import           Control.Lens
+import           Control.Monad
+import           Control.Monad.Reader
+import qualified Data.Map as M
+import           Data.Maybe hiding (mapMaybe)
+import           Data.Text (Text)
+import qualified Data.Text as T
+import           Data.Text.Lazy (fromStrict)
+import qualified Data.Text.Lazy.Encoding as T
+import           Data.Text.Read (decimal)
+import           GHCJS.DOM.Types (MonadJSM)
+import           Obelisk.Route
+import           Obelisk.Route.Frontend
+import           Reflex.Dom.Core hiding (Value)
+import           Reflex.Network
+------------------------------------------------------------------------------
+import           Chainweb.Api.ChainId
+import           ChainwebData.AccountDetail
+import           Common.Route
+import           Common.Types
+import           Common.Utils
+import           Frontend.Accounts
+import           Frontend.App
+import           Frontend.AppState
+import           Frontend.ChainwebApi
+import           Frontend.Common
+import           Frontend.Page.Block
+import           Frontend.Transactions
+------------------------------------------------------------------------------
+
+transferSearchPage
+  :: ( MonadApp r t m
+     , Prerender js t m
+     , MonadJSM (Performable m)
+     , HasJSContext (Performable m)
+     , RouteToUrl (R FrontendRoute) m
+     , SetRoute t (R FrontendRoute) m
+     , MonadIO m
+     )
+  => App AccountParams t m ()
+transferSearchPage = do
+  r <- askRoute
+  void $ networkView $ transferHelper <$> r
+
+transferHelper
+  :: ( MonadApp r t m
+     , DomBuilder t m
+     , MonadJSM (Performable m)
+     , HasJSContext (Performable m)
+     , Prerender js t m
+     , RouteToUrl (R FrontendRoute) m
+     , SetRoute t (R FrontendRoute) m
+     , MonadIO m
+     )
+  => AccountParams
+  -> App r t m ()
+transferHelper ap = transferWidget (apAccount ap) (apToken ap) (apChain ap) Nothing
+
+transferWidget
+  :: ( MonadApp r t m
+     , DomBuilder t m
+     , MonadJSM (Performable m)
+     , HasJSContext (Performable m)
+     , Prerender js t m
+     , Reflex t
+     , RouteToUrl (R FrontendRoute) m
+     , SetRoute t (R FrontendRoute) m
+     , MonadIO m
+     )
+  => Text
+  -> Text
+  -> Maybe Integer
+  -> Maybe Int
+  -> App r t m ()
+transferWidget account token chainid fromheight = do
+  (AppState n si _ _) <- ask -- TODO: add chainweb-data host to appstate record
```

Comment:
We already have [that information](https://github.com/kadena-io/block-explorer/blob/2fd8233aa999598efa5ebf1d95689d0c389d3e60/common/src/Common/Types.hs#L73) in the `_as_netConfig` field of `AppState`.